src/qt/android/src/org/bitcoincore/qt/BitcoinQtActivity.java

```diff
@@ -25,5 +27,65 @@ public void onCreate(Bundle savedInstanceState)
         }
 
         super.onCreate(savedInstanceState);
+
+        // Marshmallow+ Permission APIs
+        if (Build.VERSION.SDK_INT >= 23) {
+             fuckMarshMallow();
+         }
+    }
+
+@Override
+public void onRequestPermissionsResult(int requestCode, String[] permissions, int[] grantResults) {
+    switch (requestCode) {
+        case REQUEST_CODE_ASK_MULTIPLE_PERMISSIONS: {
+            Map<String, Integer> perms = new HashMap<String, Integer>();
+            perms.put(Manifest.permission.ACCESS_FINE_LOCATION, PackageManager.PERMISSION_GRANTED);
```

Comment:
Why do we need `ACCESS_FINE_LOCATION`?

Reply:
Ope! This is my mistake, We don't need to this permission! I'll remove this page n the next commit.